client/scss/settings/_variables.scss

```diff
@@ -106,21 +46,153 @@ $draftail-editor-z-index: $nav-wrapper-inner-z-index + 1;
 $object-title-height: 40px;
 
 // Nav
-$nav-grey-1: darken($color-white, 80);
-$nav-grey-2: darken($color-white, 60);
-$nav-item-hover-bg: rgba(100, 100, 100, 0.15);
-$nav-item-active-bg: darken($color-white, 90);
-$nav-submenu-bg: darken($color-white, 85);
-$nav-footer-account-bg: $nav-item-active-bg;
-$nav-footer-submenu-bg: $nav-submenu-bg;
 $nav-footer-closed-height: 50px;
 $nav-footer-submenu-height: 77px;
 $nav-footer-open-height: $nav-footer-closed-height + $nav-footer-submenu-height;
 
-// Nav search
-$nav-search-color: darken($color-white, 20);
-$nav-search-border: darken($color-white, 40);
-$nav-search-bg: $nav-grey-1;
-$nav-search-hover-bg: $nav-item-hover-bg;
-$nav-search-focus-color: $color-white;
-$nav-search-focus-bg: $nav-item-hover-bg;
+
+@function css-darken($hsl-tuple, $darken-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, $s, calc(#{$l} - #{$darken-by + 0%}));
+}
+@function css-lighten($hsl-tuple, $lighten-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, $s, calc(#{$l} + #{$lighten-by + 0%}));
+}
+@function css-saturate($hsl-tuple, $saturate-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, calc(#{$s} + #{$saturate-by + 0%}), $l);
+}
+@function css-desaturate($hsl-tuple, $desaturate-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, calc(#{$s} - #{$desaturate-by + 0%}), $l);
+}
+@function css-adjust-hue($hsl-tuple, $adjust-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return (calc(#{$h} + #{$adjust-by}), $s, $l);
+}
+@function css-transparentize($hsl-tuple, $alpha) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, $s, $l, $alpha);
+}
+
+// $color is either a color or an hsl tuple
+@mixin define-color($name, $color) {
+    $h: null;
+    $s: null;
+    $l: null;
+
+    @if type-of($color) == color {
+        $h: hue($color) / 1deg; // Cast to unitless
+        $s: saturation($color);
+        $l: lightness($color);
+    } @else {
+        $h: nth($color, 1);
+        $s: nth($color, 2);
+        $l: nth($color, 3);
+    }
+
+    --color-#{$name}-hue: #{$h};
+    --color-#{$name}-saturation: #{$s};
+    --color-#{$name}-lightness: #{$l};
+    --color-#{$name}: hsl(#{ var(--color-#{$name}-hue), var(--color-#{$name}-saturation), var(--color-#{$name}-lightness) });
+}
+
+@function get-color($name) {
+    @return (var(--color-#{$name}-hue), var(--color-#{$name}-saturation), var(--color-#{$name}-lightness));
+}
+
+:root {
+    // colours
+    // Defines: --color-primary-hue, --color-primary-saturation, --color-primary-lightness, and --color-primary
```

Comment:
This is a general point, not specifically about this comment – do we really need to create those four separate variables? It feels to me like it would be better to just refactor existing colors to not rely on modifying the darkness/lightness/saturation. Otherwise we end up with 20 shades that are all slightly different.

I can see how the above would be useful for transparency, but that’s about it?

Reply:
It's really nice being able to change one color and have other colors change in coordination with it... but ditching that that it would certainly make all this MUCH simpler.

Reply:
Yeah, I’m a bit torn about this. It feels neat for people to be able to change one "root" color and have other shades derived from it, but I don’t see anyone wanting to change say only the hue.